src/sentry/search/snuba/backend.py

```diff
@@ -147,16 +149,27 @@ def owner_filter(owner, projects):
                 is_active=True,
             ).values("team")
         )
-        relevant_owners = GroupOwner.objects.filter(
+        owned_by_me = GroupOwner.objects.filter(
             project_id__in=project_ids, organization_id=organization_id
-        )
-        filter_query = Q(team__in=teams) | Q(user=owner)
-        return Q(
-            id__in=relevant_owners.filter(filter_query)
+        ).filter(Q(team__in=teams) | Q(user=owner))
+        ids_assigned_to_someone_else = (
+            GroupAssignee.objects.filter(
+                project_id__in=[p.id for p in projects],
+                group_id__in=owned_by_me.values_list("group_id", flat=True).distinct(),
+            )
+            .exclude(user=owner)
             .values_list("group_id", flat=True)
             .distinct()
+        )
+        owned_by_me_and_not_assigned = owned_by_me.exclude(
+            group_id__in=ids_assigned_to_someone_else
+        )
+        return Q(
+            id__in=owned_by_me_and_not_assigned.values_list("group_id", flat=True).distinct()
         ) | assigned_to_filter(owner, projects)
     elif isinstance(owner, list) and owner[0] == "me_or_none":
+        from sentry.models.groupassignee import GroupAssignee
```

Comment:
Does this cause a cycle if it is at the top of the module?

Reply:
Nah, it doesn't....so I'll go ahead and just move it there since it's now being used more frequently.